hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/TestAvroSchemaResolutionSupport.scala

```diff
@@ -826,11 +810,7 @@ class TestAvroSchemaResolutionSupport extends HoodieClientTestBase with ScalaAss
     upsertData(df7, tempRecordPath)
 
     // read out the table
-    val readDf = spark.read.format("hudi")
-      // NOTE: type promotion is not supported for the custom file format and the filegroup reader
-      //       HUDI-7045 and PR#10007 in progress to fix the issue
-      .option(HoodieReaderConfig.FILE_GROUP_READER_ENABLED.key(), "false")
-      .load(tempRecordPath)
+    val readDf = spark.read.format("hudi").load(tempRecordPath)
```

Comment:
Can we have the schema evolution tests to be parameterized on using the new file group reader or the existing spark legacy Hudi parquet file format classes?  Both read paths should work, in case the user would like to fall back to existing way of reading the table with schema evolution.